metarecord/views/classification.py

```diff
@@ -18,27 +21,59 @@ def include_related(request):
 
 class ClassificationSerializer(serializers.ModelSerializer):
     id = serializers.UUIDField(source='uuid', format='hex', read_only=True)
-    parent = HexRelatedField(read_only=True)
+    parent = ClassificationRelationSerializer(required=False)
+    modified_by = serializers.SerializerMethodField()
 
     class Meta:
         model = Classification
-        fields = ('id', 'created_at', 'modified_at', 'code', 'title', 'parent', 'description', 'description_internal',
-                  'related_classification', 'additional_information', 'function_allowed')
+        fields = (
+            'id',
+            'created_at',
+            'modified_at',
+            'modified_by',
+            'version',
+            'state',
+            'valid_from',
+            'valid_to',
+            'code',
+            'title',
+            'parent',
+            'description',
+            'description_internal',
+            'related_classification',
+            'additional_information',
+            'function_allowed',
+        )
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         request = self.context['request']
-        self.include_related = include_related(request)
+        self.include_related = False
+
+        if request.method == 'GET':
+            self.include_related = include_related(request)
 
     def get_fields(self):
+        user = self.context['request'].user
         fields = super().get_fields()
 
         if self.include_related:
             fields['phases'] = serializers.SerializerMethodField(method_name='_get_phases')
 
+        if not user.has_perm(Classification.CAN_VIEW_MODIFIED_BY):
+            del fields['modified_by']
+
         return fields
 
+    def get_modified_by(self, obj):
+        user = self.context['request'].user
+
+        if user.has_perm(Classification.CAN_VIEW_MODIFIED_BY):
+            return obj.get_modified_by_display()
+
+        return None
```

Comment:
Is this method actually needed, isn't `modified_by` deleted from the fields in `get_fields()` anyway when the user doesn't have the permission?

Reply:
Oh yeah. I added the deletion to `get_fields` afterwards and didn't come back to this. I'll fix this right up.